kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ElementAtEvaluator.java

```diff
@@ -101,20 +102,31 @@ public boolean isNullAt(int rowId) {
             @Override
             public String getString(int rowId) {
                 lookupValue(rowId);
-                return lastLookupValue == null ? null : (String) lastLookupValue;
+                return lastLookupValue == null ? null : lastLookupValue;
             }
 
             private Object lookupValue(int rowId) {
                 if (rowId == lastLookupRowId) {
                     return lastLookupValue;
                 }
-                // TODO: this needs to be updated after the new way of accessing the complex
-                //  types is merged.
                 lastLookupRowId = rowId;
                 String keyValue = lookupKey.getString(rowId);
-                lastLookupValue = map.getMap(rowId).get(keyValue);
+                lastLookupValue = findValueForKey(map.getMap(rowId), keyValue);
                 return lastLookupValue;
             }
+
+            private String findValueForKey(MapValue mapValue, String key) {
```

Comment:
can the `key` be `null`? We should add method docs explaining the expectations on input / output, instead of requiring the reader to review the if statement below.